src/fv/share/tdyugdm.c

```diff
@@ -0,0 +1,477 @@
+#include <petscdm.h>
+#include <petscdmda.h>
+#include <petscviewerhdf5.h>
+#include <private/tdyugdmimpl.h>
+#include <private/tdymemoryimpl.h>
+#include <private/tdyutils.h>
+
+/// Allocates memory and initializes a TDyUGDM struct
+///
+/// @param [out] ugdm A TDyUGDM struct
+///
+/// @returns 0 on success, or a non-zero error code on failure
+PetscErrorCode TDyUGDMCreate(TDyUGDM **ugdm){
+  PetscFunctionBegin;
+  PetscErrorCode ierr;
+
+  ierr = PetscCalloc(sizeof(TDyUGDM), ugdm); CHKERRQ(ierr);
```

Comment:
`calloc` and friends zero out the initialized memory, so we don't need lines 19-35.